glib/src/gstring.rs

```diff
@@ -404,6 +404,29 @@ impl<'a> ToGlibPtr<'a, *mut c_char> for GString {
     }
 }
 
+#[doc(hidden)]
+impl<'a> FromGlibContainer<*const c_char, *const i8> for GString {
+    unsafe fn from_glib_none_num(ptr: *const i8, num: usize) -> Self {
+        if num == 0 {
+            return Self::from("");
+        }
+        let mut bytes = Vec::with_capacity(num);
```

Comment:
```suggestion
        let mut bytes = Vec::with_capacity(num + 1);
```

For the 0 :)

Reply:
Otherwise you'd allocate `num` bytes, copy the string, append `0` and that appending would allocate another memory area on the heap of at least `num+1` bytes, copy everything first, then free the old memory, then append the `0`.

Reply:
Interesting, thanks as always!